### PR TITLE
[PIR][DynamicShape] Add symbolic shape infer for interpolate ops

### PR DESCRIPTION
**Pull request by zhangboSJTU**
### PR Category

Others 

### PR Types

Others 

### Description

Pcard-67164

Add symbolic shape infer for interpolate ops

Although it is NOT a prime op, it will not be decomposed and generate code with CINN compiler, so this PR  is needed

Due to DimExpr can NOT act like float-type、New symbol created while floor、ceil、cast occur

### CODE REVIEW

**Review by zyfncg — APPROVED** [review]

File: paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/multiary_infer_sym.cc [hunk 1]
```diff
@@ -147,11 +313,22 @@ bool LinspaceOpInferSymbolicShape(
...
+
 bool LogspaceOpInferSymbolicShape(
     pir::Operation *op, pir::ShapeConstraintIRAnalysis *shape_analysis) {
   return LinspaceOpInferSymbolicShape(op, shape_analysis);
 }
 
+bool NearestInterpOpInferSymbolicShape(
+    pir::Operation *op, pir::ShapeConstraintIRAnalysis *shape_analysis) {
+  return BicubicInterpOpInferSymbolicShape(op, shape_analysis);
+}
```
Comment (zyfncg) [inline paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/multiary_infer_sym.cc:330]: `BicubicInterpOpInferSymbolicShape` 可以换成一个公共函数，Bicubic和其他Linear、Nearest在语义上是不同的处理策略

**Review by lanxianghit — APPROVED** [review]

File: paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/infer_sym_utils.h [hunk 2]
```diff
@@ -82,8 +87,10 @@ inline ExprVec GetExprVecFromData(const ShapeOrData &shapeordata) {
     TensorListExprs list =
         shapeordata.dyn_cast<symbol::TensorListShapeOrDataDimExprs>();
     for (size_t i = 0; i < list.size(); i++) {
-      for (auto expr : list[i].data().value()) {
-        result.emplace_back(expr);
+      if (list[i].data().has_value()) {
+        for (auto expr : list[i].data().value()) {
```
Comment (tc20042008) [inline paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/infer_sym_utils.h:91]: const auto&

Comment (zhangboSJTU) [reply paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/infer_sym_utils.h:91]: 收到、后续 PR 修正一下

File: paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/multiary_infer_sym.cc (hunk 1 above)
Comment (zhangboSJTU) [reply paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/multiary_infer_sym.cc:330]: OK、shape 推导其实是一致的、后续 PR 可以修正一下

**Pull request merged**